src/upload-helper.ts

```diff
@@ -159,10 +158,23 @@ export async function expandGlob(
   directoryPath: string,
   glob: string,
 ): Promise<string[]> {
-  const pth = toPosixPath(path.posix.join(directoryPath, glob));
-  const filesList = await globby([pth], {
+  // Preserve backwards compatability and evaluate the empty glob to a
+  // recursive include-all.
+  if (!glob) {
+    glob = '**/*';
+  }
+
+  // tinyglob requires a posix path, but then the rest of our code wants a
+  // OS-specific path.
+  const pth = toPosixPath(path.join(directoryPath, glob));
+  const filesList = await tinyglob(pth, {
     dot: true,
+    filesOnly: true,
+    flush: true,
   });
+  for (let i = 0; i < filesList.length; i++) {
```

Comment:
It looks like this was the major change. `tinyglob` wants a posix path, but it _returns_ an OS-path. So we have to normalize the result back to what `globby` was doing, which was always returning a posix-style path. Otherwise we'd have to redo all our upstream logic.